July news (#1101)
ropensci/roweb3: content/blog/2025-07-23-news-july-2025/index.Rmd

### rOpenSci at useR! 2025

August 1, [useR! 2025 Virtual](https://user2025.r-project.org/program/virtual/) with **free registration**. 

August 08-10, 2025 [useR! 2025](https://user2025.r-project.org) will take place in Penn Pavilion, Duke University, Durham, North Carolina. 

**rOpenSci will be there!**

* [Yanina Bellini Saibene](/author/yanina-bellini-saibene/) will be presenting her keynote ["We R Together.  How to learn, use and improve a programming language as a community."](https://user2025.r-project.org/program/in-person/keynotes/yanina-bellini-saibene) 

* [Will Landau](/author/will-landau/) will be presenting his keynote ["Poweful simulation pipelines with {targets}"](https://user2025.r-project.org/program/in-person/keynotes/will-landau) 

* [Noam Ross](/author/noam-ross/) will be presenting "Curating a Community of Packages: Lessons from a Decade of rOpenSci Peer Review" on Sat, Aug 9, 2025 - 13:00–14:10 (EDT)

* Our Champion [Erika Siregar](/author/erika-siregar/) is presenting "rPlaywright: Bringing Playwright’s Power to R for Scalable Web Automation", Fri, Aug 8, 2025 - 18:15–19:30 (EDT)

> * [Andrea Gomez Vargas](/author/andrea-gomez-vargas/), one of our Champions, is presenting "ARcenso: A Package Born from Chaos, Powered by Community" on Sat, Aug 9, 2025 - 13:00–14:10 (EDT)

Author: ```suggestion
* Our Champion [Andrea Gomez Vargas](/author/andrea-gomez-vargas/) is presenting "ARcenso: A Package Born from Chaos, Powered by Community" on Sat, Aug 9, 2025 - 13:00–14:10 (EDT)
```

### Cache values with automatic pruning: the cachem package

If you're writing a wrapper for an API that has no regular caching headers, 
you might be interested in the [cachem package](https://github.com/r-lib/cachem/) for caching.
Thanks to Tan Ho and Pieter Huybrechts for the tip.
See its use in [comtradr](https://github.com/ropensci/comtradr/blob/f5b2715bf6dddcaff039c3613e8f4ecb0607e5c0/R/zzz.R#L58) by Paul Bochtler.

### A debugging tip

Miles McBain shared an interesting debugging tip in his post [Dive()ing into the hunt #rstats](https://milesmcbain.micro.blog/2025/07/04/diveing-into-the-hunt-rstats.html).

### How to prepare examples in your package for a CRAN submission

The CRAN cookbook by Jasmine Daly and Beni Altmann  features a section called ["Structuring of Examples"](https://contributor.r-project.org/cran-cookbook/general_issues.html#structuring-of-examples) mentioning dontrun and friends and their use cases.
Thanks to Hugo Gruson for the find!

### Don't ask "any updates on this?" on GitHub

> Carlos Scheidegger, a developer of Quarto, wrote an interesting [Bluesky thread](https://bsky.app/profile/cscheid.net/post/3ls25wg6mnk2y) about why it's not considerate to write "any update on this?" on open bug reports or feature requests.
> 
> > 'But "any updates?" just makes everyone feel bad.''

Reviewer: I'd suggest adding the important bit (or what I think is the important bit), which is what to do _instead_. It's a bit more positive and actionable.

```suggestion
Carlos Scheidegger, a developer of Quarto, wrote an interesting [Bluesky thread](https://bsky.app/profile/cscheid.net/post/3ls25wg6mnk2y) about why it's not considerate to write "any update on this?" on open bug reports or feature requests.

Carlos suggests that better alternatives are to "add _new_ information to the post about how you want to use it, or other workarounds" which might give developers ideas for different fixes, or to simply upvote the request. But avoid "any updates?", which is hard on the developer because all they can say is "no", and they know that will disappoint people. 
```

* Our Champion [Erika Siregar](/author/erika-siregar/) is presenting "rPlaywright: Bringing Playwright’s Power to R for Scalable Web Automation", Fri, Aug 8, 2025 - 18:15–19:30 (EDT)

* Our Champion [Andrea Gomez Vargas](/author/andrea-gomez-vargas/) is presenting "ARcenso: A Package Born from Chaos, Powered by Community" on Sat, Aug 9, 2025 - 13:00–14:10 (EDT)

* [Luis D. Verde Arregoita](/author/luis-d.-verde-arregoitia/) is presenting "Bringing the fun of hex stickers to your R session" on Sun, Aug 10, 2025 - 10:30–12:00 (EDT)

We are looking forward to seeing you there!

### Coworking 

Read [all about coworking](/blog/2023/06/21/coworking/)!

- Tuesday August 5th, 2025 9:00 AM Australia Western (01:00 UTC), ["Drawing publication-ready plots using R"](/events/coworking-2025-08/) with [Steffi LaZerte](/author/steffi-lazerte/) and cohost [Emi Tanaka](/author/emi-tanaka/).
    - Work on some figures and plots for your own work;
    - Explore ways of making publication-ready plots in R;
    - Chat with Emi Tanaka about making publication-ready plots in R.

> - Tuesday September 9th, 14:00 European Central (12:00 UTC), ["Improving accessibility in your work](/events/coworking-2025-09/) with [Steffi LaZerte](/author/steffi-lazerte/) and cohost [Silvia Canelón](/author/silvia-canel%C3%B3n/).

Reviewer: ```suggestion
- Tuesday September 9th, 14:00 European Central (12:00 UTC), ["Improving accessibility in your work"](/events/coworking-2025-09/) with [Steffi LaZerte](/author/steffi-lazerte/) and cohost [Silvia Canelón](/author/silvia-canel%C3%B3n/).
```

    - Explore where you might improve accessibility in your work;
    - Start making changes to improve accessibility;

>     - Chat with Silvia Canelón about accessibility.

Reviewer: ```suggestion
    - Chat with Silvia Canelón about tools and techniques for improving accessibility.
```

>     

Reviewer: ```suggestion
    
- Tuesday October 7th, 09:00 Americas Pacific (16:00 UTC), ["Ship your R package to the R-multiverse!"](/events/coworking-2025-10/) with [Steffi LaZerte](/author/steffi-lazerte/) and cohost [Will Landau](/author/will-landau).
    - Learn more about the [R-multiverse](https://r-multiverse.org/);
    - Submit your package to the [R-multiverse](https://r-multiverse.org/);
    - Chat with Will Landau, and learn more about the R-multiverse and how to submit your package.
    
```
